### PR TITLE
[21771] Spent time section is shown even if there is no time logged

### PR DESCRIPTION
**Pull request by furinvader**
If the `spentTime` field has the value `"0 hours"`, hide it.

https://community.openproject.org/work_packages/21771/activity

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://furinvader-bugs-21771-spent-time-visible.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://furinvader-bugs-21771-spent-time-visible.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: frontend/app/work_packages/services/work-package-field-service.js [hunk 1]
```diff
@@ -218,8 +218,8 @@ module.exports = function(
       return true;
     }
 
-    // strategy pattern, guys
-    if (field === 'spentTime' && WorkPackageFieldService.getValue(workPackage, field) === 'PT0S') {
+    if (field === 'spentTime' && WorkPackageFieldService.getValue(workPackage, field) === 'PT0S'
+        || value === '0 hours') {
```
Comment (basiszwo) [inline frontend/app/work_packages/services/work-package-field-service.js]: I think you cannot compare to `0 hours` as this string maybe localized and you will get `0 Stunden` or sth. else. Check by changing your personal language setting in the personal profile.

Comment (furinvader) [inline frontend/app/work_packages/services/work-package-field-service.js]: Thanks for your remark. I'm on it.

**Comment by oliverguenther** [comment]
@furinvader The second solution doesn't work properly, as well. It hides the spent time always, even when it's duration is non-zero as `getValue` seems to always be null in your case.

I've fixed and tested this over at https://github.com/opf/openproject/pull/3734, which uses the raw duration value from the work package as before, but uses the correct location to retrieve it.

**Pull request closed without merging**